x-pack/plugins/reporting/public/lib/job.tsx

```diff
@@ -169,45 +186,45 @@ export class Job {
   }
 
   getWarnings() {
-    if (this.status !== FAILED) {
-      const warnings: string[] = [];
-      if (this.isDeprecated) {
-        warnings.push(
-          i18n.translate('xpack.reporting.jobWarning.exportTypeDeprecated', {
-            defaultMessage:
-              'This is a deprecated export type. Automation of this report will need to be re-created for compatibility with future versions of Kibana.',
-          })
-        );
-      }
-      if (this.csv_contains_formulas) {
-        warnings.push(
-          i18n.translate('xpack.reporting.jobWarning.csvContainsFormulas', {
-            defaultMessage:
-              'Your CSV contains characters that spreadsheet applications might interpret as formulas.',
-          })
-        );
-      }
-      if (this.max_size_reached) {
-        warnings.push(
-          i18n.translate('xpack.reporting.jobWarning.maxSizeReachedTooltip', {
-            defaultMessage: 'Your search reached the max size and contains partial data.',
-          })
-        );
-      }
-
-      if (this.warnings?.length) {
-        warnings.push(...this.warnings);
-      }
-
-      if (warnings.length) {
-        return (
-          <ul>
-            {warnings.map((w, i) => {
-              return <li key={`warning-key-${i}`}>{w}</li>;
-            })}
-          </ul>
-        );
-      }
+    const warnings: string[] = [];
+    if (this.isDeprecated) {
+      warnings.push(
+        i18n.translate('xpack.reporting.jobWarning.exportTypeDeprecated', {
+          defaultMessage:
+            'This is a deprecated export type. Automation of this report will need to be re-created for compatibility with future versions of Kibana.',
+        })
+      );
+    }
+
+    if (this.csv_contains_formulas) {
+      warnings.push(
+        i18n.translate('xpack.reporting.jobWarning.csvContainsFormulas', {
+          defaultMessage:
+            'Your CSV contains characters that spreadsheet applications might interpret as formulas.',
+        })
+      );
+    }
+    if (this.max_size_reached) {
+      warnings.push(
+        i18n.translate('xpack.reporting.jobWarning.maxSizeReachedTooltip', {
+          defaultMessage: 'Your search reached the max size and contains partial data.',
+        })
+      );
+    }
+
+    // warnings could contain the failure message
+    if (this.status !== FAILED && this.warnings?.length) {
+      warnings.push(...this.warnings);
```

Comment:
If the job has failed, then the only possible warning that we will show is the deprecation warning.

If it is not deprecated, we only show the error message related to the failure - this because the error message text is provided in the `warnings` field in the job data.

If it has not failed, we show all the warnings.